samples/client/petstore/c/api/PetAPI.c

```diff
@@ -10,6 +10,83 @@
     char dst[256];\
     snprintf(dst, 256, "%ld", (long int)(src));\
 }while(0)
+
+
+
+
+
+
+
+
+// Functions for enum STATUS for PetAPI_findPetsByStatus
+
```

Comment:
Hello @michelealbano @zhemant 

Actually, for the test case Petstore, the function involve enum is:

```
list_t*
PetAPI_findPetsByStatus(apiClient_t *apiClient ,list_t * status);
```
The status is a list rather than an enum, so it does not need add the code in api-header and api-body.

yaml snippet of [petstore.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/2_0/petstore.yaml)
```
parameters:
        - name: status
          in: query
          description: Status values that need to be considered for filter
          required: true
          type: array
          items:
            type: string
            enum:
              - available
              - pending
              - sold
```

Maybe we need another template code to control whether generating the enum related code in api header and body.

Reply:
But this is only for enum, what if there is a struct. Even the structs are not generated from the path parameters. I remember I was told that other generators support creating models from path params, I think we should do same. So, in this case, a model will be generated for enum and it will be handled as a list of models in api-body . If we have two api and they have same struct defined in path params, then it should be generated as a model rather than putting model code in api-files

Reply:
Hi @zhemant  

I agree with you that the enum in API path should be created in api-body for this case.  

But I have a little question:

Take the petstore as an example,  there is an enum "pet_status_e" created by model-body, it can be used for 
```
list_t*
PetAPI_findPetsByStatus(apiClient_t *apiClient ,list_t * status);
```

 User can use pet_status_e for the "status" of PetAPI_findPetsByStatus

There is no need to create 
```
typedef enum  {  STATUS_available, STATUS_pending, STATUS_sold } status_e;
```
in api-body.

It is duplicated but I admit it is not a big deal.

Reply:
Should I do something for this thread? Please try out the C generator as it is now, and tell me if you like the result.

Reply:
Hello @michelealbano 

I don't understand why the generated functions (findPetsByStatus_STATUS_ToString, openapi_petstore_findPetsByStatus_status_e ) are "static" ?  (because I found they are not used in this source file, and users cannot use these functions in their source codes )

Actually, I agree with your commit first time, though it is duplicated for petstore sample, but it is acceptable. 

And why need add "{{#allowableValues}}" ?

Reply:
Hi,
I made the functions "static" since I did not want to fill up the namespace too much. If they were not static, they would have to be heavily decorated. By the way, I see no use cases where we need *_convertToJSON and *_parseFromJSON for the enums, since the JSONification is done on the whole models. Should I remove enums' *_convertToJSON and *_parseFromJSON from the api-body, like I did with the model-body?

I added `{{#allowableValues}}` since the models' dump is:
```
      "allowableValues" : {
        "enumVars" : [ {
          "name" : "NOTSECURE",
          "isString" : false,
          "value" : "NOT_SECURE"
        }, {
          "name" : "CERTIFICATE",
          "isString" : false,
          "value" : "CERTIFICATE"
        }, {
          "name" : "TOKEN",
          "isString" : false,
          "value" : "TOKEN"
        } ],
        "values" : [ "NOT_SECURE", "CERTIFICATE", "TOKEN" ]
      }```
Thus, I had to go "inside" allowableValues to retrieve the possible values for the enums.

Reply:
Yes. I think you can if they are not needed for now.

> Should I remove enums' *_convertToJSON and *_parseFromJSON from the api-body, like I did with the model-body?

Reply:
Done